sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyUtils.java

```diff
@@ -177,6 +196,37 @@ public static List<HttpRequest> getSanitizerRequests(List<TestProxySanitizer> sa
         }).collect(Collectors.toList());
     }
 
+    public static List<HttpRequest> getMatcherRequests(List<TestProxyMatcher> matchers) {
+        return matchers.stream().map(testProxyMatcher -> {
+            HttpRequest request;
+            String matcherType;
+            switch (testProxyMatcher.getType()) {
+                case HEADERLESS:
+                    matcherType = TestProxyMatcher.TestProxyMatcherType.HEADERLESS.getName();
+                    request
+                        = new HttpRequest(HttpMethod.POST, String.format("%s/Admin/setmatcher", TestProxyUtils.getProxyUrl()));
+                    break;
+                case BODILESS:
+                    request
+                        = new HttpRequest(HttpMethod.POST, String.format("%s/Admin/setmatcher", TestProxyUtils.getProxyUrl()));
+                    matcherType = TestProxyMatcher.TestProxyMatcherType.BODILESS.getName();
+                    break;
+                case CUSTOM:
+                    CustomMatcher customMatcher = (CustomMatcher) testProxyMatcher;
+                    String requestBody = createCustomMatcherRequestBody(customMatcher);
+                    matcherType = TestProxyMatcher.TestProxyMatcherType.CUSTOM.getName();
+                    request
+                        = new HttpRequest(HttpMethod.POST, String.format("%s/Admin/setmatcher", TestProxyUtils.getProxyUrl())).setBody(requestBody);
+                    break;
+                default:
+                    throw new RuntimeException(String.format("Matcher type {%s} not supported", testProxyMatcher.getType()));
+            }
+
+            request.setHeader("x-abstraction-identifier", matcherType);
+            return request;
+        }).collect(Collectors.toList());
+    }
+
```

Comment:
It occurs to me that the pattern here, instead of using an enum for a type, is to have some `abstract` function on this base type that just creates a `HttpRequest`?

Reply:
this isn't a now problem but is a refactoring we should consider.